### PR TITLE
Adding support for exporting/importing labels to/from Scale AI

### PR DESCRIPTION
**Pull request by brimoor**
Adds support for importing labels in [Scale AI format](https://docs.scale.com/reference#general-image-annotation), as well as exporting labels in a format that can be uploaded as pre-annotations via Scale AI tasks' `hypothesis` parameter.

All of the following label types are supported:

Images
- [General image annotation](https://docs.scale.com/reference#general-image-annotation), including boxes, polylines, polygons, and keypoints
- [Semantic segmentation](https://docs.scale.com/reference#semantic-segmentation-annotation)

Videos
- [General video annotation](https://docs.scale.com/reference#general-video-annotation)
- [Video playback](https://docs.scale.com/reference#video-playback)

Here's an example of exporting BDD labels into Scale AI format and then importing them back into the same dataset using the API introduced in this PR:

See `tests/misc/scale_tests.py` for additional tests.

```py
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.utils.scale as fous

SCALE_EXPORT_PATH = "/tmp/scale-image-export.json"
SCALE_IMPORT_PATH = "/tmp/scale-image-import.json"
SCALE_ID_FIELD = "scale_id"

_dataset = foz.load_zoo_dataset("bdd100k", split="validation", shuffle=True, max_samples=10)
dataset = fo.Dataset()
dataset.add_samples(_dataset.take(10))

# Export labels in Scale format
fous.export_to_scale(
    dataset,
    SCALE_EXPORT_PATH,
    label_prefix="",  # all fields
)

# Convert to Scale import format
id_map = fous.convert_scale_export_to_import(
    SCALE_EXPORT_PATH, SCALE_IMPORT_PATH
)

for sample_id, task_id in id_map.items():
    sample = dataset[sample_id]
    sample[SCALE_ID_FIELD] = task_id
    sample.save()

# Import labels from Scale
fous.import_from_scale(
    dataset,
    SCALE_IMPORT_PATH,
    label_prefix="scale",
    scale_id_field=SCALE_ID_FIELD,
)

# Verify that we have two copies of the same labels
session = fo.launch_app(dataset)
```

Screen Shot 2020-11-09 at 9 07 32 PM

### CODE REVIEW

File: fiftyone/utils/scale.py [hunk 1]
```diff
@@ -0,0 +1,1077 @@
...
+                "annotations": [...],
+                "events": [...],
+            }
+
+    Args:
+        dataset: a :class:`fiftyone.core.dataset.Dataset`
+        labels_dir_or_json: the path to a Scale AI JSON export or a directory
+            of JSON exports as per the formats described above
+        label_prefix (None): a prefix to prepend to the sample label field(s)
+            that are created
```
Comment (benjaminpkane) [inline fiftyone/utils/scale.py:161]:
```suggestion
            that are created, separated by an underscore
```

File: fiftyone/utils/scale.py [hunk 2]
```diff
@@ -0,0 +1,1077 @@
...
+
+        {
+            "annotations": [...],
+            "events": [...],
+        }
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+            the path to write an NDJSON export of the labels
```
Comment (benjaminpkane) [inline fiftyone/utils/scale.py:318]: typo?

File: fiftyone/utils/scale.py [hunk 3]
```diff
@@ -0,0 +1,1077 @@
...
+
+            labels[sample.id] = {
+                "filepath": sample.filepath,
+                "hypothesis": anno_dict,
+            }
+
+    etas.write_json(labels, json_path)
+
+
+def convert_scale_export_to_import(inpath, outpath):
```
Comment (benjaminpkane) [inline fiftyone/utils/scale.py:425]: This is only for testing/demonstrative purposes, correct? Perhaps that could be explicitly noted in the doc string.

**Review by benjaminpkane — APPROVED** [review]
LGTM! It is very exciting to see this. I think it is clear line in the sand that has been crossed by introducing these annotation provider integrations.

I by no means clicked through and referenced every Scale link to check this work. Taking their API on in one go is very impressive to say the least, though. :rocket:

File: fiftyone/utils/scale.py (hunk 3 above)
Comment (brimoor) [reply fiftyone/utils/scale.py:425]: Mostly, but it could be used to provide Importer/Exporter-like functionality for sharing labels in Scale format

**Pull request merged**